custom_components/car_wash/binary_sensor.py

```diff
@@ -157,7 +161,15 @@ async def async_update(self):
         tmpu = self.hass.config.units.temperature_unit
         temp = wdata.attributes.get(ATTR_WEATHER_TEMPERATURE)
         cond = wdata.state
-        forecast = wdata.attributes.get(ATTR_FORECAST)
+
+        daily_response = await self.hass.services.async_call(
+            WEATHER_DOMAIN,
+            SERVICE_GET_FORECASTS,
+            {"entity_id": self._weather_entity, "type": "daily"},
+            blocking=True,
+            return_response=True,
+        )
+        forecast = daily_response[self._weather_entity]["forecast"]
```

Comment:
Please add a check for the presence of a result in the return value (daily_response).  And also add a unit test for your code.

Reply:
Thank you for the review and requested changes.
I've just added a new commit, with the changes which are looking good to me.

Also changed some imports, because ruff suggested these changes.